Add pre-commit config

ericholscher opened the pull request:
<!-- readthedocs-preview writethedocs-www start -->
----
📚 Documentation preview 📚: https://writethedocs-www--2483.org.readthedocs.build/

<!-- readthedocs-preview writethedocs-www end -->

ericholscher committed:
Add pre-commit config

plaindocs commented:
Very much of a fan of these. Do they need any local installation? Or do they 'just work'?

ericholscher commented:
Yea you need to install the pre commit package. Should probably add that to the README.

ericholscher commented:
@copilot Add this to the requirements and readme.

plaindocs reviewed (approved):
Should probably merge this. 

```
pre-commit run --all-files
```

Doesn't work imo, but is less important than changed files

ericholscher committed:
Fix pre-commit hooks for YAML schema validation

- Exclude schema files from check-yaml hook (they use multi-document YAML format)
- Exclude schema files from yamllint hook
- Fix yamale hooks to properly handle file arguments using bash wrapper
- Change yamale hooks to use system language instead of python
- Exclude schema files from being validated against themselves

This resolves all pre-commit validation errors.

ericholscher committed:
Fix linting issues

ericholscher commented:
@plaindocs Think I have it passing locally now.

ericholscher commented:
```
vscode ➜ /workspaces/projects/www (pre-commit) $ ~/.local/bin/pre-commit run --all-files
check yaml...............................................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
yamllint.................................................................Passed
Validate conference config YAML against schema...........................Passed
Validate schedule YAML against schema....................................Passed
Validate sessions YAML against schema....................................Passed
```
